crates/pathfinder/src/storage/schema/revision_0018.rs

```diff
@@ -0,0 +1,202 @@
+use anyhow::Context;
+use rusqlite::Transaction;
+
+/// Primary goal of this migration is to allow for StarkNet block forks.
+///
+/// This is achieved by re-creating the `starknet_blocks` table, shifting the PK
+/// from `number` to `hash`. This in turn requires re-creating any tables with a FK
+/// on the original `starknet_blocks` table.
+///
+/// In addition, a `canonical_blocks` table is created to track the canonical block chain
+/// now that forks are allowed in `starknet_blocks`.
+pub(crate) fn migrate(tx: &Transaction<'_>) -> anyhow::Result<()> {
+    tx.execute(
+        r"-- Stores StarkNet block headers.
+CREATE TABLE starknet_blocks_new (
+    hash      BLOB    PRIMARY KEY NOT NULL,
+    number    INTEGER NOT NULL,
+    root      BLOB    NOT NULL,
+    timestamp INTEGER NOT NULL, 
+    gas_price BLOB    NOT NULL,
+    sequencer_address BLOB NOT NULL,
+    version_id INTEGER REFERENCES starknet_versions(id)
```

Comment:
These fields had default values specified because they were added after table creation. I've dropped the default values here.

Reply:
Okay so I apparently misunderstood this. I thought you could now ignore these values as they would have the default value of the original table. This is however not the case -- the migration fails instead.

Will add the defaults back in I guess.

Reply:
pebkak error.. I had forgotten to include those fields in the insert. Defaults are not needed anymore.